src/index.js

```diff
@@ -20,12 +20,30 @@ const client = new ApolloClient({
   cache: new InMemoryCache(),
 });
 
-ReactDOM.render(
-  <ApolloProvider client={client}>
-    <App />
-  </ApolloProvider>,
-  document.getElementById('root')
-);
+function renderApp(config) {
+  ReactDOM.render(
+    <ApolloProvider client={client}>
+      <App {...config} />
+    </ApolloProvider>,
+    document.getElementById('root')
+  );
+}
+
+fetch('config.json')
+  .then(response => {
+    if (!response.ok) {
+      throw Error('Got ' + response.status + ' http status');
```

Comment:
Does the `fetch()` function return a proper ES6 promise? I think those would route this exception to the `.catch`, whereas older solutions, at least jQuery's promises, would not.

Reply:
There is a test case added that shows it's the case.